iothub/service/src/Messaging/MessagesClient.cs

```diff
@@ -407,9 +410,28 @@ private void OnConnectionClosed(object sender, EventArgs e)
             }
         }
 
-        private void CheckConnectionIsOpen()
+        private async Task CheckConnectionIsOpenAsync(CancellationToken cancellationToken = default)
         {
-            if (!_amqpConnection.IsOpen)
+            if (_opened)
```

Comment:
I d recommend to add a comment here and state that this is to make sure the `OpenAsync` call was made before but the client was disconnected afterwards.